Formula/asdf.rb

```diff
@@ -4,29 +4,17 @@ class Asdf < Formula
   url "https://github.com/asdf-vm/asdf/archive/v0.8.0.tar.gz"
   sha256 "9b667ca135c194f38d823c62cc0dc3dbe00d7a9f60caa0c06ecb3047944eadfa"
   license "MIT"
-  revision 1
+  revision 2
   head "https://github.com/asdf-vm/asdf.git"
 
   bottle :unneeded
 
-  depends_on "autoconf"
-  depends_on "automake"
-  depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
-  depends_on "openssl@1.1"
-  depends_on "readline"
-  depends_on "unixodbc"
-
-  conflicts_with "homeshick",
-    because: "asdf and homeshick both install files in lib/commands"
-
   def install
     bash_completion.install "completions/asdf.bash"
     fish_completion.install "completions/asdf.fish"
     zsh_completion.install "completions/_asdf"
-    libexec.install "bin/private"
-    prefix.install Dir["*"]
+    libexec.install Dir["*"]
+    bin.write_exec_script (libexec/"bin/asdf")
     touch prefix/"asdf_updates_disabled"
```

Comment:
Should this be
```ruby
touch libexec/"asdf_updates_disabled"
```
?

Reply:
@carabolic
Actually, I think that entire line is unnecessary. 
Just to confirm, you are *correct* that it should be there, good catch.
But my suggestion is to remove it. 

Again if @jthegedus could confirm, but it looks like it also checks
`|| ! git rev-parse --is-inside-work-tree &>/dev/null; then`
Which will obviously fail because we aren't a git repo. 

I think that's a better approach. So I'll remove this, but if you or @jthegedus feel differently, I will revert.

Reply:
That wasn't the case from #72419, so maybe @cho-m can weigh in here.

Reply:
@carlocab That wasn't the case because you were installing lib and bin without. Try it yourself, just install this formula as a tap.

Reply:
@seivan I agree that if the ASDF install location is no longer inside Homebrew's git repo, then it should be fine to remove.

ASDF's self-update is based on cloning the asdf git repo, so the fact that it was located inside Homebrew's git repo was confusing the update scripts.

Reply:
@cho-m @carlocab I am however curious over what happens when you install using `HEAD` or if someone were to change into using `:git` with `url "https://github.com/asdf-vm/asdf.git", :using => :git, :tag => "v0.8.0" `

Maybe it's not a bad idea after all to add it, in case something changes down the line?

Reply:
So it seems like even using `git` with defined tags, it doesn't remain as a git repo on its own. 
Either way, this isn't necessary.

The update command is defined as 
```shell
update_command() {
  local update_to_head=$1

  (
    cd "$(asdf_dir)" || exit 1

    if [ -f asdf_updates_disabled ] || ! git rev-parse --is-inside-work-tree &>/dev/null; then
      echo "Update command disabled. Please use the package manager that you used to install asdf to upgrade asdf."
      exit 42
    else
      do_update "$update_to_head"
    fi
  )
}
```

It cds into `asdf_dir` and runs git, and since it's not a git repo, it will fail.

Reply:
@seivan taking another look, one thing to note is that `HOMEBREW_PREFIX == HOMEBREW_REPOSITORY` on M1 macs: https://github.com/Homebrew/install/blob/dee8df98bfb65588007c666034c6e1ad0733b1b6/install.sh#L36

`HOMEBREW_REPOSITORY` is a git repo, https://github.com/Homebrew/install/blob/dee8df98bfb65588007c666034c6e1ad0733b1b6/install.sh#L647

So, everything under the prefix directory is a Git repo on M1 ARM macs.

ARM location:
```console
❯ eval $(/opt/homebrew/bin/brew shellenv)

❯ cd $HOMEBREW_PREFIX

❯ pwd
/opt/homebrew

❯ git remote get-url origin
https://github.com/Homebrew/brew
```

Intel location:
```console
❯ eval $(/usr/local/bin/brew shellenv)

❯ cd $HOMEBREW_PREFIX

❯ pwd
/usr/local

❯ git remote get-url origin
fatal: not a git repository (or any of the parent directories): .git

❯ cd Homebrew

❯ git remote get-url origin
https://github.com/Homebrew/brew
```

Reply:
@cho-m Good catch! I'll revert it!
Thanks again, I haven't paid much attention to other archs.

Reply:
It's been fixed now and CI shows green, so I'll resolve this.